plugins/processors/ifname/ifname.go

```diff
@@ -0,0 +1,328 @@
+package ifname
+
+import (
+	"fmt"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/internal/snmp"
+	si "github.com/influxdata/telegraf/plugins/inputs/snmp"
+	"github.com/influxdata/telegraf/plugins/processors"
+	"github.com/influxdata/telegraf/plugins/processors/reverse_dns/parallel"
+)
+
+var sampleConfig = `
+  ## Name of tag holding the interface number
+  # tag = "ifIndex"
+
+  ## Name of output tag where service name will be added
+  # dest = "ifName"
+
+  ## Name of tag of the SNMP agent to request the interface name from
+  # agent = "agent"
+
+  ## Timeout for each request.
+  # timeout = "5s"
+
+  ## SNMP version; can be 1, 2, or 3.
+  # version = 2
+
+  ## SNMP community string.
+  # community = "public"
+
+  ## Number of retries to attempt.
+  # retries = 3
```

Comment:
can we default to zero retries?

Reply:
This is another case of copying the snmp input.  I thought it would be best to have the same behavior there and here but I'm open to changing it there's something wrong with it.

Reply:
See the last comment on https://github.com/influxdata/telegraf/issues/7300